src/class.ziprecipes.php

```diff
@@ -1785,5 +1779,76 @@ public static function zrdn_format_image($item) {
         }
         return $item;
     }
-}
 
+    /**
+     * Get Responsive Image attributes from URL
+     * 
+     * It checks image is not external and return images attributes like srcset, sized etc.
+     * 
+     * @param type $url
+     * @return type
+     */
+    public static function zrdn_get_responsive_image_attributes($url) {
+        /**
+         * set up default array values
+         */
+        $attributes = [];
+        $attributes['url'] = $url;
+        $attributes['attachment_id'] = $attachment_id = attachment_url_to_postid($url);
+        $attributes['srcset'] = '';
+        $attributes['sizes'] = '';
+        if ($attachment_id) {
+            $attributes['url'] = wp_get_attachment_image_url($attachment_id, 'full');
+            $image_meta = wp_get_attachment_metadata($attachment_id);
+            // $attributes['meta'] = esc_attr($image_meta); // may need in future for alt, meta title
+            $img_srcset = wp_get_attachment_image_srcset($attachment_id, 'full', $image_meta);
```

Comment:
This is returning `false` for me even though I have ImageMagick extension.

```
In media.php, the place where this check happens

     if ( empty( $image_meta['sizes'] ) || ! isset( $image_meta['file'] ) || strlen( $image_meta['file'] ) < 4 ) {
		return false;
	}

`$image_meta` looks like this:

‌array (
  'width' => 1102,
  'height' => 915,
  'file' => '2018/03/fruit-sorbet-cake.jpg',
  'image_meta' => 
  array (
    'aperture' => '0',
    'credit' => '',
    'camera' => '',
    'caption' => '',
    'created_timestamp' => '0',
    'copyright' => '',
    'focal_length' => '0',
    'iso' => '0',
    'shutter_speed' => '0',
    'title' => '',
    'orientation' => '1',
    'keywords' => 
    array (
    ),
  ),
)
```
```

Reply:
You need to regenerate Thumbnails. You can use any plugin for this purpose